File: src/main/java/br/anhembi/funmodechild/controller/CategoryController.java
package br.anhembi.funmodechild.controller;

import br.anhembi.funmodechild.model.response.CategoryResponse;
import br.anhembi.funmodechild.service.CategoryService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(value = "/categories", produces = MediaType.APPLICATION_JSON_VALUE)
@Tag(name = "Categories")
public class CategoryController {

    private final CategoryService categoryService;

    public CategoryController(CategoryService categoryService) {
        this.categoryService = categoryService;
    }

    @GetMapping
    @Operation(summary = "Get a list of available categories")
    public List<CategoryResponse> getCategories() {
        return categoryService.getCategories();
    }
}

File: src/main/kotlin/br/anhembi/funmodechild/controller/CategoryController.kt
package br.anhembi.funmodechild.controller

import br.anhembi.funmodechild.model.response.CategoryResponse
import br.anhembi.funmodechild.service.CategoryService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/categories"], produces = [MediaType.APPLICATION_JSON_VALUE])
@Tag(name = "Categories")
class CategoryController(private val categoryService: CategoryService) {

    @GetMapping
    @Operation(summary = "Get a list of available categories")
    fun listCategories(): List<CategoryResponse> = categoryService.listCategories()
}
